### PR TITLE
feat: added incarnation two

### PR DESCRIPTION
**Pull request by LucasChollet**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "python.analysis.extraPaths": [
```
Comment (LucasChollet) [inline .vscode/settings.json:2]: Looks very related to your configuration. I'm not sure why we should have it in the repo.

File: src/incarnation_two.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import argparse
+import sys
+import sympy as sp
+
+
+class CoasterOverlap:
```
Comment (LucasChollet) [inline src/incarnation_two.py:6]: The class and all methods need documentation, even for `init`.

File: src/incarnation_two.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    def __init__(self, radius):
+        self.R = radius
+        self.alpha = self.compute_alpha()
+        self.overlap_length = self.compute_overlap_length()
+
+    def compute_alpha(self):
```
Comment (LucasChollet) [inline src/incarnation_two.py:12]: Can we have some type hinting?
`-> float` or `-> None` as trailing return type would be cool.

File: src/incarnation_two.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        """Computes the angle alpha based on the given equation."""
```
Comment (LucasChollet) [inline src/incarnation_two.py:13]: Please conserve the style that is already used elsewhere, with new line after and before `"""`
Also, a `:return:` tag is lacking

File: src/incarnation_two.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        alpha = sp.Symbol('alpha')
+        equation = alpha - sp.sin(alpha) - sp.pi / 2
+        alpha_solution = sp.nsolve(equation, alpha, 2)
+        return alpha_solution
+
+    def compute_overlap_length(self):
+        """Computes the length of the overlapping region."""
+        l = 2 * self.R * (1 - sp.cos(self.alpha / 2))
```
Comment (LucasChollet) [inline src/incarnation_two.py:21]: Same for stack variables, static typing is much appreciated:
```suggestion
        l: float = 2 * self.R * (1 - sp.cos(self.alpha / 2))
```

File: test/test_incarnation_two.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import unittest
```
Comment (LucasChollet) [inline test/test_incarnation_two.py:1]: This file needs documentation on class/methods too.
Also, (and I forgot on the other file), docstrings on top of the file are required.

File: test/test_incarnation_two.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        ]
+
+        for radius, expected_overlap_length in test_data:
+            with self.subTest(radius=radius, expected_overlap_length=expected_overlap_length):
+                coaster_overlap = CoasterOverlap(radius)
+                overlap_length = coaster_overlap.compute_overlap_length()
+                self.assertAlmostEqual(overlap_length, expected_overlap_length, places=7)
+
+
+if __name__ == '__main__':
```
Comment (LucasChollet) [inline test/test_incarnation_two.py:29]: Why is that included, `unittests` are intended to be called from the commad line.

**Review by LucasChollet — COMMENTED** [review]
Also, please squash your commits, no one wants to see your `.pyc` files haha

**Comment by LucasChollet** [comment]
Ohh and tests are failing because `sympy` is not in the environment, please add that to this file [environment.yml](https://github.com/LucasChollet/SOEN-6441/blob/main/environment.yml)

File: src/incarnation_two.py (hunk 3 above)
Comment (Alor-e) [reply src/incarnation_two.py:12]: oh yeah sorry I forgot about type hints

File: src/incarnation_two.py (hunk 2 above)
Comment (Alor-e) [reply src/incarnation_two.py:6]: On this I have the documentation, but I didn't include it as discussed.

File: test/test_incarnation_two.py (hunk 6 above)
Comment (Alor-e) [reply test/test_incarnation_two.py:1]: I have that too but did not push it as discussed

File: test/test_incarnation_two.py (hunk 7 above)
Comment (Alor-e) [reply test/test_incarnation_two.py:29]: Of course the unittests are run from the command line but this is included in case of a specific run.

**Comment by Alor-e** [comment]
> Ohh and tests are failing because `sympy` is not in the environment, please add that to this file [environment.yml](https://github.com/LucasChollet/SOEN-6441/blob/main/environment.yml)

Alright I will do this

File: src/incarnation_two.py (hunk 2 above)
Comment (Alor-e) [reply src/incarnation_two.py:6]: Okay I misunderstood you, I think you meant to say docstring (not documentation as in pydoc documentation)

**Comment by Alor-e** [comment]
All tests have passed.
